system-test/logging-winston.ts

```diff
@@ -97,55 +88,174 @@ describe('LoggingWinston', () => {
-              });
-
-              done();
+          });
+        },
+      },
+      {
+        args: [new Error('forth')],
+        level: 'error',
+        verify: (entry: types.StackdriverEntry) => {
+          assert((entry.data as {
+                   message: string
+                 }).message.startsWith('forth Error:'));
+        },
+      },
+      {
+        args: [{
+          level: 'error',
+          message: 'fifth message',
+          error: new Error('fifth')
+        }],
+        level: 'log',
+        verify: (entry: types.StackdriverEntry) => {
+          assert((entry.data as {
+                   message: string
+                 }).message.startsWith('fifth message'));
+
+          // assert(entry!.data!.metadata!.error,'Error: fifth Error:'
+        },
+      },
+    ];
+
+    const LOG_NAME = 'logging_winston_3_system_tests';
+    const LoggingWinston = inject('../src/index', {
+                             winston: winston3,
+                             'winston/package.json': {version: '3.0.0'}
+                           }).LoggingWinston;
+
+    const logger = winston3.createLogger({
+      transports: [new LoggingWinston({logName: LOG_NAME})],
+    });
+
+    it('should properly write log entries', (done) => {
+      const start = Date.now();
+      testData.forEach((test) => {
+        // logger does not have index signature.
+        // tslint:disable-next-line:no-any
+        (logger as any)[test.level].apply(logger, test.args);
+      });
+
+      pollLogs(LOG_NAME, start, testData.length, WRITE_CONSISTENCY_DELAY_MS)
+          .then((entries) => {
+            assert.strictEqual(entries.length, testData.length);
+            entries.reverse().forEach((entry, index) => {
+              const test = testData[index];
+              test.verify(entry);
             });
-      }, WRITE_CONSISTENCY_DELAY_MS);
+            done();
+          });
     });
   });
 
+
   describe('ErrorReporting', () => {
+    const LoggingWinston = inject('../src/index', {
+                             winston: winston2,
+                             'winston/package.json': {version: '2.2.0'}
+                           }).LoggingWinston;
+
+    const logger = new winston2.Logger({
+      transports: [new LoggingWinston({
+        logName: LOG_NAME,
+        serviceContext:
+            {service: 'logging-winston-system-test', version: 'none'}
+      })],
+    });
+
     const ERROR_REPORTING_DELAY_MS = 10 * 1000;
     const errorsTransport = new ErrorsApiTransport();
 
     beforeEach(async function() {
       this.timeout(2 * ERROR_REPORTING_DELAY_MS);
+
       await errorsTransport.deleteAllEvents();
       await new Promise((resolve, reject) => {
         setTimeout(resolve, ERROR_REPORTING_DELAY_MS);
       });
     });
 
-    afterEach(async () => {
-      await errorsTransport.deleteAllEvents();
-    });
+    afterEach(
+        async () => {
+            // await errorsTransport.deleteAllEvents();
```

Comment:
Is this commented-out line intended?

Reply:
it was but ill add it back as an `after` its a call to be tidy not really for the test because its also called in beforeEach